### PR TITLE
added canonicalUrl field for article

### PR DESCRIPTION
**Pull request by bishoymelek**
this PR is to add new field in article metadata panel for the canonical URL to improve SEO

### CODE REVIEW

**Review by HanaAyman — APPROVED** [review]

**Review by penk85 — DISMISSED** [review]

File: examples/website/src/shared/route/articleTemplateContainer.tsx [hunk 1]
```diff
@@ -129,20 +129,21 @@ export function ArticleTemplateContainer({id, slug}: ArticleTemplateContainerPro
     socialMediaDescription,
     socialMediaImage,
     socialMediaAuthors,
-    comments
+    comments,
+    canonicalUrl
   } = articleData
 
   const path = ArticleRoute.reverse({id, slug})
-  const canonicalURL = canonicalHost + path
```
Comment (roosnic1) [inline examples/website/src/shared/route/articleTemplateContainer.tsx:136]: I'd keep `canonicalURL` and add a second const called `canonicalPeerURL`

File: examples/website/src/shared/route/articleTemplateContainer.tsx [hunk 2]
```diff
@@ -129,20 +129,21 @@ export function ArticleTemplateContainer({id, slug}: ArticleTemplateContainerPro
...
     <>
       <Helmet>
         <title>{title}</title>
         {lead && <meta name="description" content={lead} />}
-        <link rel="canonical" href={canonicalURL} />
+        <link rel="canonical" href={articleCanonicalURL} />
         <meta property="og:title" content={socialMediaTitle ?? title} />
         <meta property="og:type" content="article" />
-        <meta property="og:url" content={canonicalURL} />
+        <meta property="og:url" content={canonicalHost + path} />
```
Comment (roosnic1) [inline examples/website/src/shared/route/articleTemplateContainer.tsx:146]: you can keep `canonicalURL` here

File: examples/website/src/shared/route/articleTemplateContainer.tsx [hunk 3]
```diff
@@ -160,9 +161,9 @@ export function ArticleTemplateContainer({id, slug}: ArticleTemplateContainerPro
         <meta name="twitter:card" content="summary_large_image"></meta>
       </Helmet>
 
-      <DesktopSocialMediaButtons shareUrl={canonicalURL} />
+      <DesktopSocialMediaButtons shareUrl={articleCanonicalURL} />
```
Comment (roosnic1) [inline examples/website/src/shared/route/articleTemplateContainer.tsx:164]: you can keep `canonicalURL` here

File: examples/website/src/shared/route/articleTemplateContainer.tsx [hunk 4]
```diff
@@ -160,9 +161,9 @@ export function ArticleTemplateContainer({id, slug}: ArticleTemplateContainerPro
...
       <BlockRenderer
-        articleShareUrl={canonicalURL}
+        articleShareUrl={articleCanonicalURL}
```
Comment (roosnic1) [inline examples/website/src/shared/route/articleTemplateContainer.tsx:166]: you can keep `canonicalURL` here

File: examples/website/src/shared/route/articleTemplateContainer.tsx [hunk 5]
```diff
@@ -274,19 +275,20 @@ export function PeerArticleTemplateContainer({
   } = articleData
 
   const path = PeerArticleRoute.reverse({peerID: '12', id, slug})
-  const canonicalURL = canonicalHost + path
+
+  const peerArticleCanonicalURL = articleData.canonicalUrl || articleData.url
```
Comment (roosnic1) [inline examples/website/src/shared/route/articleTemplateContainer.tsx:279]: i'd keep `canonicalURL`

File: examples/website/src/shared/route/articleTemplateContainer.tsx [hunk 6]
```diff
@@ -274,19 +275,20 @@ export function PeerArticleTemplateContainer({
...
         <title>{title}</title>
         {lead && <meta name="description" content={lead} />}
 
-        <link rel="canonical" href={canonicalURL} />
+        <link rel="canonical" href={peerArticleCanonicalURL} />
 
         <meta property="og:title" content={socialMediaTitle ?? title} />
         <meta property="og:type" content="article" />
-        <meta property="og:url" content={canonicalURL} />
+        <meta property="og:url" content={canonicalHost + path} />
```
Comment (roosnic1) [inline examples/website/src/shared/route/articleTemplateContainer.tsx:291]: you can keep `canonicalURL` here

File: examples/website/src/shared/route/articleTemplateContainer.tsx [hunk 7]
```diff
@@ -304,10 +306,10 @@ export function PeerArticleTemplateContainer({
         ))}
       </Helmet>
 
-      <DesktopSocialMediaButtons shareUrl={canonicalURL} />
+      <DesktopSocialMediaButtons shareUrl={peerArticleCanonicalURL} />
```
Comment (roosnic1) [inline examples/website/src/shared/route/articleTemplateContainer.tsx:309]: you can keep `canonicalURL` here

File: examples/website/src/shared/route/articleTemplateContainer.tsx [hunk 8]
```diff
@@ -304,10 +306,10 @@ export function PeerArticleTemplateContainer({
...
       <PeerProfileBlock peer={peer} article={articleData} />
       <BlockRenderer
-        articleShareUrl={canonicalURL}
+        articleShareUrl={peerArticleCanonicalURL}
```
Comment (roosnic1) [inline examples/website/src/shared/route/articleTemplateContainer.tsx:312]: you can keep `canonicalURL` here

File: packages/editor/src/client/panel/articleMetadataPanel.tsx [hunk 9]
```diff
@@ -381,6 +383,14 @@ export function ArticleMetadataPanel({
                   onChange={breaking => onChange?.({...value, breaking})}
                 />
               </FormGroup>
+              <FormGroup>
+                <ControlLabel>{t('articleEditor.panels.canonicalUrl')}</ControlLabel>
+                <FormControl
+                  value={canonicalUrl}
+                  onChange={canonicalUrl => onChange?.({...value, canonicalUrl})}
+                />
+                <HelpBlock>{t('articleEditor.panels.canonicalUrLHelpBlock')}</HelpBlock>
```
Comment (roosnic1) [inline packages/editor/src/client/panel/articleMetadataPanel.tsx:392]: Help block should contain a link to https://developers.google.com/search/docs/advanced/crawling/consolidate-duplicate-urls

**Review by roosnic1 — CHANGES_REQUESTED** [review]

File: examples/website/src/shared/route/articleTemplateContainer.tsx (hunk 1 above)
Comment (bishoymelek) [reply examples/website/src/shared/route/articleTemplateContainer.tsx:136]: @roosnic1 check the updates please

File: examples/website/src/shared/route/articleTemplateContainer.tsx [hunk 10]
```diff
@@ -134,16 +134,18 @@ export function ArticleTemplateContainer({id, slug}: ArticleTemplateContainerPro
   } = articleData
 
   const path = ArticleRoute.reverse({id, slug})
-  const articleCanonicalURL = canonicalUrl || canonicalHost + path
+  const canonicalURL = canonicalHost + path
+  const canonicalPeerURL = canonicalUrl || canonicalHost + path
```
Comment (roosnic1) [inline examples/website/src/shared/route/articleTemplateContainer.tsx:138]: this is very bad variable naming. You have a `canonicalURL` and a `canonicalUrl`. Please change this.

File: examples/website/src/shared/route/articleTemplateContainer.tsx [hunk 11]
```diff
@@ -276,19 +278,20 @@ export function PeerArticleTemplateContainer({
 
   const path = PeerArticleRoute.reverse({peerID: '12', id, slug})
 
-  const peerArticleCanonicalURL = articleData.canonicalUrl || articleData.url
+  const canonicalURL = canonicalHost + path
+  const canonicalPeerURL = articleData.canonicalUrl || articleData.url
```
Comment (roosnic1) [inline examples/website/src/shared/route/articleTemplateContainer.tsx:282]: same as above

**Review by roosnic1 — CHANGES_REQUESTED** [review]

File: examples/website/src/shared/route/articleTemplateContainer.tsx (hunk 10 above)
Comment (bishoymelek) [reply examples/website/src/shared/route/articleTemplateContainer.tsx:138]: @roosnic1 good point, I changed it to be canonicalOwnURL. What do you think?

File: examples/website/src/shared/route/articleTemplateContainer.tsx (hunk 11 above)
Comment (bishoymelek) [reply examples/website/src/shared/route/articleTemplateContainer.tsx:282]: @roosnic1 answered above.

**Review by roosnic1 — APPROVED** [review]

**Pull request merged**